DisguiseUnityRenderStream/Runtime/Util/MulticastExtensions.cs

```diff
@@ -0,0 +1,130 @@
+            // multiple nodes locally and unit testing.
+            // Food for thought: Does it have a performance cost? Do we want to have it configurable or disabled in some
+            //                   cases?
+            udpClient.Client.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, true);
+
+            // Bind to receive from the selected adapter on the same port than the port we are sending to (everyone will
+            // use the same port).
+            udpClient.Client.Bind(new IPEndPoint(adapterAddress, port));
+
+            // Join the multicast group
+            udpClient.JoinMulticastGroup(multicastIPAddress);
+
+            // This is normally true by default but this is required to keep things simple (unit tests working, multiple
+            // instances on the same computer, ...).  So let's get sure it is on.
+            udpClient.MulticastLoopback = true;
+        }
+
+        static (NetworkInterface, UnicastIPAddressInformation) SelectNetworkInterface(string adapterName)
+        {
+            var nics = NetworkInterface.GetAllNetworkInterfaces();
+            NetworkInterface firstUpNic = null;
+            UnicastIPAddressInformation firstUpNicIPAddress = null;
+
+            foreach (var nic in nics)
+            {
+                bool isExplicitNic = !string.IsNullOrEmpty(adapterName) && nic.Name == adapterName;
+                bool isUp = nic.OperationalStatus == OperationalStatus.Up;
+
+                if (!isUp)
+                {
+                    if (isExplicitNic)
+                    {
+                        Debug.LogError(
+                            $"Unable to use explicit interface: \"{nic.Name}\", the interface is down. Attempting to use the next available interface.");
+                    }
+
+                    continue;
+                }
+
+                var ipProperties = nic.GetIPProperties();
+                if (ipProperties == null || !ipProperties.MulticastAddresses.Any())
+                {
+                    continue;
+                }
+
+                if (!nic.SupportsMulticast)
+                {
+                    continue;
+                }
+
+                UnicastIPAddressInformation nicIPAddress = null;
+                foreach (var ip in nic.GetIPProperties().UnicastAddresses)
+                {
+                    if (ip.Address.AddressFamily != AddressFamily.InterNetwork)
+                        continue;
+
+                    nicIPAddress = ip;
+                    break;
+                }
+
+                if (nicIPAddress == null)
+                {
+                    continue;
+                }
+
+                if (IPAddress.IsLoopback(nicIPAddress.Address))
+                {
+                    // Skip loopback adapter as they cause all sort of problems with multicast...
+                    continue;
+                }
+
+                firstUpNic ??= nic;
```

Comment:
Brainstorm: If I remember correctly this part of code is quite similar to what we are doing in CD to automatically select a network adapter to use for state propagation.  However, I wonder if we could do a little bit better for Disguise.

From what I understand render stream nodes will have at least 2 NIC up and running:
1. For transmission of video signal
2. General networking

#1 will most likely be a high performance & reliable network, especially if using uncompressed video, it will be a network of at least 10 gbps, most likely 25 and it could event be 100 gbps.  Even better, it will be configured for nice reliable udp multicast (they are using SMPTE 2110 under the hood that is using udp multicast).
#2 will most likely be any network and we have no idea how it will be configured, how it is reliable, ...

So based on that I think we should probably try to use #1.  And the easiest way to identify it would be to use the one with the highest speed (as opposed to the first one we find as this code is currently doing).

Reply:
Do you know of the API to get the speed of the NIC reliably? There's `NetworkInterface.Speed`, but I don't know if that's the preferred way.

Reply:
AFAIK it returns what the NIC driver reports to windows, so that's probably good enough.

Reply:
Probably a good idea to exclude loopback and wireless adapters using NetworkInterface.NetworkInterfaceType